n3fit/src/evolven3fit_new/utils.py

```diff
@@ -123,15 +178,22 @@ def get_theoryID_from_runcard(usr_path):
     return my_runcard["theory"]["theoryid"]
 
 
-def generate_q2grid(Q0, Qfin, Q_points, match_dict):
+def generate_q2grid(Q0, Qfin, Q_points, match_dict, nf0=None):
     """Generate the q2grid used in the final evolved pdfs or use the default grid if Qfin or Q_points is
     not provided.
 
     match_dict contains the couples (mass : factor) where factor is the number to be multiplied to mass
     in order to obtain the relative matching scale.
     """
     if Qfin is None and Q_points is None:
-        return DEFAULT_Q2GRID
+        if nf0 == 4:
```

Comment:
Since now it is becoming more difficult and we have two different defaults, I am wondering if we should just remove all the defaults and just compute the q2grid with Q0, Qfin and Q_points

Reply:
Given a theory I think it makes sense to have a default or at least a recipe. Since the theories are a bit messy right now I think I prefer to have two defaults for the ~400 theories we have rather than put my hands there (for now).